### PR TITLE
Command for moving nodes between namespaces

### PR DESCRIPTION
**Pull request by Kazauwa**
Added command for moving nodes between namespaces (see [#362](https://github.com/juanfont/headscale/issues/362))

- [x] read the [CONTRIBUTING guidelines](README.md#user-content-contributing)
- [x] raised a GitHub issue or discussed it on the projects chat beforehand
- [x] added unit tests
- [x] added integration tests
- [ ] updated documentation if needed
- [x] updated CHANGELOG.md

### CODE REVIEW

File: cmd/headscale/cli/nodes.go [hunk 1]
```diff
@@ -46,6 +46,18 @@ func init() {
...
+	err = moveNodeCmd.MarkFlagRequired("identifier")
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	moveNodeCmd.Flags().StringP("namespace", "n", "", "New namespace")
+	err = moveNodeCmd.MarkFlagRequired("namespace")
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	nodeCmd.AddCommand(moveNodeCmd)
```
Comment (kradalby) [inline cmd/headscale/cli/nodes.go:60]:
```suggestion
	moveNodeCmd.Flags().Uint64P("identifier", "i", 0, "Node identifier (ID)")
	
	err = moveNodeCmd.MarkFlagRequired("identifier")
	if err != nil {
		log.Fatalf(err.Error())
	}
	
	moveNodeCmd.Flags().StringP("namespace", "n", "", "New namespace")
	
	err = moveNodeCmd.MarkFlagRequired("namespace")
	if err != nil {
		log.Fatalf(err.Error())
	}
	nodeCmd.AddCommand(moveNodeCmd)
```

**Review by kradalby — COMMENTED** [review]
Looks good, can you have a look at adding some integration senario as well? `integration_cli_test.go`

File: cmd/headscale/cli/nodes.go (hunk 1 above)
Comment (Kazauwa) [reply cmd/headscale/cli/nodes.go:60]: > Looks good, can you have a look at adding some integration senario as well? `integration_cli_test.go`
On it!

**Comment by Kazauwa** [comment]
Alright, integration tests were lifesavers. Turns out that a machine's namespace wasn't updating properly.

File: integration_cli_test.go [hunk 2]
```diff
@@ -1222,3 +1222,149 @@ func (s *IntegrationCLITestSuite) TestApiKeyCommand() {
...
+			"json",
+		},
+		[]string{},
+	)
+	assert.Nil(s.T(), err)
+
+	err = json.Unmarshal([]byte(moveToNewNSResult), &machine)
+	assert.Nil(s.T(), err)
+
+	assert.Equal(s.T(), machine.Namespace, newNamespace)
```
Comment (kradalby) [inline integration_cli_test.go:1301]: Could you yank the `nodes list` code from one of the other tests just to verify that we can read it back correctly after move?

**Review by kradalby — COMMENTED** [review]
Great!, one tiny extra safety, but otherwise great!

File: integration_cli_test.go (hunk 2 above)
Comment (Kazauwa) [reply integration_cli_test.go:1301]: Sure, no problems!

Comment (Kazauwa) [reply integration_cli_test.go:1301]: Done

**Review by kradalby — APPROVED** [review]

**Pull request merged**